bin/git-jira-branch.rb

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env ruby
+# encoding: UTF-8
+#
+# git-jira-branch
+# 
+#   Suggest a branch name from the given JIRA issue ID
+# 
+#   Assumes the branches are named as:
+#   <team>/<branch-title>-<story-id>
+# 
+require 'rubygems'
+require 'optparse'
+require 'jira'
+require 'readline'
+require 'term/ansicolor'
+require 'git-whistles/app'
+
+class App < Git::Whistles::App
+
+  def initialize
+    super
+  end
+
+  def main(args)
+    super
+    parse_args!(args)
+
+    if args.count < 1
+      show_and_exit usage
+    end
+
+    story_id = args[0]
+
+    # get PT info
+    issue = get_jira_info(story_id)
+
+    branch_name_suggested = "#{story_id.upcase}-#{issue.fields['summary'].downcase}"
+    branch_name_suggested.gsub!(/[^\w\d\/]/, '-').gsub!(/-+/, '-')
+
+    puts 'The suggested branch name is: ' << Term::ANSIColor.yellow(branch_name_suggested)
+    puts '    Press ENTER if you agree'
+    puts ' or Write any other name and press ENTER'
+    puts ' or Press CTRL-D to cancel'
+
+    branch_name = Readline.readline '  > ', false
+
+    if branch_name.nil?
+      log.warn "\nCancelled by user"
+      exit 2
+    end
+
+    puts 'Would you like to transition the issue?'
```

Comment:
Not sure if that would be confusing is the numbers are out of order.  What do you think?